src/main/java/dk/cngroup/trainings/spring/springassignment/service/CareTakerServiceImpl.java

```diff
@@ -72,21 +80,15 @@ public boolean deleteCareTakerById(long id) {
 	}
 
 	@Override
-	public String addExistingAnimalToExistingCareTaker(long careTakerId, long animalId) {
-		Optional<CareTaker> careTaker = careTakerRepository.findById(careTakerId);
+	public void addExistingAnimalToExistingCareTaker(long careTakerId, long animalId)
+			throws AnimalNotFoundException, CareTakerNotFoundException, AnimalAndCareTakerAlreadyLinked {
+		Optional<CareTaker> careTaker = this.getCareTakerById(careTakerId);
 		Optional<Animal> animal = animalService.getAnimalById(animalId);
-		if (!careTaker.isPresent()) {
-			return "Failed: CareTaker id:" + careTakerId + " doesn't exist";
-		} else if (!animal.isPresent()) {
-			return "Failed: Animal id:" + animalId + " doesn't exist";
+		if (!careTaker.get().getAnimals().stream().anyMatch(a -> a.getId().equals(animalId))) {
```

Comment:
Instead of negation, I think it would be better to use noneMatch method (or what is the name :) )

Reply:
I will try it.

Reply:
Done